pkg/metadata/labels.go

```diff
@@ -41,10 +41,17 @@ const (
 	InstanceLabel          = LabelKROPrefix + "instance-name"
 	InstanceNamespaceLabel = LabelKROPrefix + "instance-namespace"
 
+	// Parent Identifier Labels. These are set by the instance reconciler that creates/updates the resource.
 	ResourceGraphDefinitionIDLabel        = LabelKROPrefix + "resource-graph-definition-id"
 	ResourceGraphDefinitionNameLabel      = LabelKROPrefix + "resource-graph-definition-name"
 	ResourceGraphDefinitionNamespaceLabel = LabelKROPrefix + "resource-graph-definition-namespace"
 	ResourceGraphDefinitionVersionLabel   = LabelKROPrefix + "resource-graph-definition-version"
+
+	// Source Identifier Labels. These are set by the instance reconciler that reconciles the resource.
+	SourceResourceGraphDefinitionIDLabel        = LabelKROPrefix + "source-resource-graph-definition-id"
```

Comment:
Maybe stupid question: in which case can we have differing labels here? I.E. is there a case in which we have a differing ResourceGraphDefinitionIDLabel and SourceResourceGraphDefinitionIDLabel

Reply:
When we have a RGD2's instance as part of another RGD1, we want to inject labels that can identify the creator of the instance (RGD1) as well as the reconciler of that instance (RGD2)